### DISCUSSION TITLE
Does watchtower work with containers started with docker-compose?

### DISCUSSION THREAD

**Discussion by Coderdude112** [containrrr/watchtower#1189] [Q&A] [answered]
Hello!

The question is quite simple, will watchtower work if I am running all my containers through docker-compose? I understand this is still docker but am not sure if it will stop watchtower from working.

Thanks!

**Comment by piksel** [comment, marked as answer]
It shouldn't cause it to stop working, but there are/were versions of `docker compose` (docker-compose v2) that didn't put the image name in the started containers. That caused watchtower to not be able to update those containers (since there was no mention of what image to look for).
I have not checked if this is still the case with the latest `docker compose` though. See the sticky #1050 .

**Reply by simonaronsson** [reply]
As in "it shouldn't **stop** working". Another thing to keep in mind is that if watchtower update a container, running `docker-compose up` again would still recreate them, as the digest hash used by compose won't be recalculated by watchtower.
